libfreerdp/core/input.c

```diff
@@ -937,6 +937,14 @@ BOOL freerdp_input_send_mouse_event(rdpInput* input, UINT16 flags, UINT16 x, UIN
 	if (freerdp_settings_get_bool(input->context->settings, FreeRDP_SuspendInput))
 		return TRUE;
 
+	/* Save cursor position in prevent-session-lock mode */
```

Comment:
* time should also be updated for keyboard events (scancode/unicode) 
* you need to filter mouse wheel events (they have `0,0` as they do not have coordinates (easiest: just check for `PTR_FLAGS_MOVE|PTR_FLAGS_BUTTON1|PTR_FLAGS_BUTTON2|PTR_FLAGS_BUTTON3`
* you also should check for extended mouse events